activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryTasksIT.java

```diff
@@ -1583,10 +1584,51 @@ public void should_getTask_when_queryFilteredByProcessDefinitionName() {
             assertThat(responseEntity.getBody()).isNotNull();
             Collection<Task> tasks = responseEntity.getBody().getContent();
             assertThat(tasks)
-                    .extracting(Task::getId)
-                    .containsExactly(task1.getId());
+                .extracting(Task::getId)
+                .containsExactly(task1.getId());
         });
 
     }
 
+    @Test
+    public void shouldFilterTasksForCompletedBy() {
+
+        //Given
+        String completedByFirstUser = "hruser1";
+        String completedBySecondUser = "userXyz";
+
+        Task assignedTask = taskEventContainedBuilder
+            .aCompletedTaskWithCompletedBy("Assigned task1",
+                runningProcessInstance, completedByFirstUser);
+        taskEventContainedBuilder
+            .aCompletedTaskWithCompletedBy("Assigned task2",
+                runningProcessInstance, completedBySecondUser);
+
+        eventsAggregator.sendAll();
+
+        await().untilAsserted(() -> {
+            //check for specific completed by value
+            //when
+            ResponseEntity<PagedModel<Task>> responseEntity = testRestTemplate
+                .exchange(TASKS_URL + "?completedBy=" + completedByFirstUser,
+                    HttpMethod.GET,
+                    keycloakTokenProducer.entityWithAuthorizationHeader(),
+                    PAGED_TASKS_RESPONSE_TYPE
+                );
+
+            //then
+            assertThat(responseEntity).isNotNull();
+            assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+            assertThat(responseEntity.getBody()).isNotNull();
+            Collection<Task> tasks = responseEntity.getBody().getContent();
+            assertThat(tasks).hasSize(1);
```

Comment:
Note: `hasSize(1)` is redundant here as you're already using `containsExactly` (which will fail if the size is different from `1`).